### PR TITLE
Feature/48 question group description

### LINKED ISSUE #48: Question Group Description
## TODO/DONE

- [x] Question Group Description
- [x] Unit test

### PR DESCRIPTION
**Pull request by ifirmawan**
## TODO/DONE

- [x] Update style question group description
- [x] Add more tests in FieldGroupHeader component

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205014844634486

### CODE REVIEW

File: app/src/form/support/FieldGroupHeader.js [hunk 1]
```diff
@@ -7,9 +7,17 @@ const FieldGroupHeader = ({ index, name = '', description = '' }) => {
   return (
     <View>
       <View style={styles.fieldGroupHeader}>
-        <Text style={styles.fieldGroupName}>{!isNaN(index) ? `${index + 1}. ${name}` : name}</Text>
+        <Text style={styles.fieldGroupName} testID="text-name">
+          {!isNaN(index) ? `${index + 1}. ${name}` : name}
+        </Text>
+      </View>
+      <View style={styles.fieldGroupDescContainer}>
+        {description && (
```
Comment (dedenbangkit) [inline app/src/form/support/FieldGroupHeader.js]: Use descriptive variable names that accurately represent the purpose of the variables.

```jsx
    {hasDescription && (
      <Text style={styles.fieldGroupDescription} testID="text-description">
        {descriptionText}
      </Text>
    )}
```

File: app/src/form/support/FieldGroupHeader.js [hunk 2]
```diff
@@ -7,9 +7,17 @@ const FieldGroupHeader = ({ index, name = '', description = '' }) => {
   return (
     <View>
       <View style={styles.fieldGroupHeader}>
-        <Text style={styles.fieldGroupName}>{!isNaN(index) ? `${index + 1}. ${name}` : name}</Text>
+        <Text style={styles.fieldGroupName} testID="text-name">
+          {!isNaN(index) ? `${index + 1}. ${name}` : name}
```
Comment (dedenbangkit) [inline app/src/form/support/FieldGroupHeader.js]: why we have `!isNaN(index)` ? this means field group headers sometimes doesn't have index?

**Review by dedenbangkit — CHANGES_REQUESTED** [review]

File: app/src/form/support/FieldGroupHeader.js (hunk 2 above)
Comment (ifirmawan) [reply app/src/form/support/FieldGroupHeader.js]: This means the Header component has an option to use the index or not, and `isNaN` for this case to handle the index is not a number.

But I think we can change the conditional statement like this
```
index ? `${parseInt(index) + 1}. ${name}` : name
```

is this Okay for you mas @dedenbangkit?

File: app/src/form/support/FieldGroupHeader.js (hunk 1 above)
Comment (ifirmawan) [reply app/src/form/support/FieldGroupHeader.js]: noted. I'll fix it

File: app/src/form/support/FieldGroupHeader.js (hunk 2 above)
Comment (ifirmawan) [reply app/src/form/support/FieldGroupHeader.js]: wait, I think an index should always exist and is a number if we look based on this use case

https://github.com/akvo/nmis-mobile/blob/main/app/src/form/components/QuestionGroup.js#L9

https://github.com/akvo/nmis-mobile/blob/main/app/src/form/FormContainer.js#L70C2

https://github.com/akvo/nmis-mobile/blob/main/app/src/form/lib/index.js#L71

Comment (dedenbangkit) [reply app/src/form/support/FieldGroupHeader.js]: Please make it **strict**. Array will always have index.

**Review by dedenbangkit — APPROVED** [review]

**Pull request merged**